lib/src/algorithms.dart

```diff
@@ -32,10 +32,10 @@ int _comparableBinarySearch/*<T extends Comparable<T>>*/(
 /// the objects.
 ///
 /// Returns -1 if [value] is not in the list by default.
-int binarySearch/*<T extends Comparable<T>>*/(
+int binarySearch/*<T>*/(
     List/*<T>*/ sortedList, /*=T*/ value, { int compare(/*=T*/ a, /*=T*/ b) }) {
   if (compare == null) {
-    return _comparableBinarySearch(sortedList, value);
+    return _comparableBinarySearch(sortedList as List<Comparable>, value as Comparable);
```

Comment:
List<dynamic> will be as List<Comparable>. How could you get from a List<Comparable> to something that isn't List<Comparable>?